src/index.js

```diff
@@ -0,0 +1,223 @@
+  const cityNameInput = document.getElementById("cityNameInput");
+  const currentCityHeader = document.getElementById("headerCityName");
+  currentCityHeader.textContent = cityNameInput.value;
+};
+
+const handleUpdateCityEvent = () => {
+  const cityNameInput = document.getElementById("cityNameInput");
+  const resetButton = document.getElementById("cityNameReset");
+  cityNameInput.addEventListener("input", updateCity);
+  resetButton.addEventListener("click", ResetCity);
+};
+  
+/************************/
+/******* Wave 4 *********/
+/************************/
+
+const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
+
+// Event handler function to retrieve user input:
+const retrieveCityInput = () => {
+  const cityNameInput = document.getElementById("cityNameInput");
+  return cityNameInput.value;
+};
+
+// API Call to LocationIQ for location coordinates
+const getCoordinates = () => {
+  let location = retrieveCityInput();
+
+  return axios
+    .get("http://127.0.0.1:5000/location", {
+      params: {
+        q: location,
+      },
+    })
+    .then((response) => {
+      let latitude = response.data[0].lat;
+      let longitude = response.data[0].lon;
+
+      console.log(
+        `Location: ${location}, Latitude: ${latitude}, Longitude: ${longitude}`
+      );
+      return {
+        latitude: latitude,
+        longitude: longitude,
+      };
+    })
+    .catch((error) => {
+      console.log("Error found inside getCoordinates function!");
+      console.log(
+        `The value of status inside of error response is: 
+        ${error.response.status}`
+      );
+    });
+};
+
+// API Call for weather status/temp of current city displayed
+const getCurrentCityWeather = () => {
+  // Info from OpenWeather API Call documentation:
+  // https://openweathermap.org/current#geo
+  return getCoordinates()
+    .then((coordinatesResponse) => {
+      if (!coordinatesResponse) {
+        throw new Error("Coordinates could not be retrieved.");
+      } else {
+        return axios.get("http://127.0.0.1:5000/weather", {
+          params: {
+            lat: coordinatesResponse.latitude,
+            lon: coordinatesResponse.longitude,
+          },
+        });
+      }
+    })
+    .then((locationTempResponse) => {
+      let temp = locationTempResponse.data.main.temp; // in KELVIN
+      console.log(`Current Location temperature is: ${temp}`);
+      return temp;
+    })
+    .catch((error) => {
+      console.log("Error found inside getCurrentCityWeather!");
+      console.log(
+        `The value of status inside of error response is: 
+        ${error}`
+      );
+    });
+};
+
+const updateRealtimeTempBtn = () => {
+  // Chaining the axios Promise API call/response from above
+  getCurrentCityWeather().then((tempKelvin) => {
+    if (tempKelvin) {
+      const conversion = (tempKelvin - 273) * (9 / 5) + 32;
+      state.currentTemp = Math.round(conversion);
+      updateTemperatureDisplay();
+    }
+  });
```

Comment:
Nice chaining! Only suggestion, moving the temperature conversion logic into its own function so that it would be more readable since we are already working inside of a `.then` chain.